### PR TITLE
fsutils/ipcfg fixes

### PR DESCRIPTION
**Pull request by davids5**
## Summary

1. fsutils:ipcfg Add extern 'C' under cpp
2. fsutils:ipcfg Fix range checking for PRTOCOL in bin mode
3. fsutils:ipcfg Fix fd propagation as return value in bin mode 

## Impact
Fixes use with C++
Fixes use with CONFIG_IPCFG_BINARY set

## Testing
px4_fmuv5x

### CODE REVIEW

File: fsutils/ipcfg/ipcfg.c [hunk 1]
```diff
@@ -220,11 +218,14 @@ static int ipcfg_open(FAR const char *netdev, int oflags, mode_t mode)
...
       ret = -errno;
+      close(fd);
       fprintf(stderr, "ERROR: Failed to seek to $ld: %d\n",
               (long)CONFIG_IPCFG_OFFSET, ret);
-      close(fd);
+      goto errout_with_path;
     }
+
 #endif
+  ret = fd;
```
Comment (patacongo) [inline fsutils/ipcfg/ipcfg.c:229]: I don't see any logic difference with this change.  What is the problem that you are solving?  Or is this just NIH?

Comment (patacongo) [reply fsutils/ipcfg/ipcfg.c:229]: Never mind.  It is clearer looking at the whole file.

Comment (davids5) [reply fsutils/ipcfg/ipcfg.c:229]: No. It's a bug.
Valid fd is 3 but code returns 0 (OK)
image

**Review by Ouss4 — APPROVED** [review]
LGTM

**Pull request merged**